server/controllers/monitorController.js

```
@@ -243,56 +244,122 @@ class MonitorController {
 	};
 
 	/**
-	 * Creates bulk monitors and adds them to the job queue.
+	 * Creates bulk monitors and adds them to the job queue after parsing CSV.
 	 * @async
 	 * @param {Object} req - The Express request object.
-	 * @property {Object} req.body - The body of the request.
+	 * @property {Object} req.file - The uploaded CSV file.
 	 * @param {Object} res - The Express response object.
 	 * @param {function} next - The next middleware function.
-	 * @returns {Object} The response object with a success status, a message indicating the creation of the monitor, and the created monitor data.
+	 * @returns {Object} The response object with a success status and message.
 	 * @throws {Error} If there is an error during the process, especially if there is a validation error (422).
 	 */
 	createBulkMonitors = async (req, res, next) => {
 		try {
-			await createMonitorsBodyValidation.validateAsync(req.body);
-		} catch (error) {
-			next(handleValidationError(error, SERVICE_NAME));
-			return;
-		}
-
-		try {
-			// create monitors
-			const monitors = await this.db.createBulkMonitors(req);
-
-			// create notifications for each monitor
-			await Promise.all(
-				monitors.map(async (monitor, index) => {
-					const notifications = req.body[index].notifications;
-
-					if (notifications?.length) {
-						monitor.notifications = await Promise.all(
-							notifications.map(async (notification) => {
-								notification.monitorId = monitor._id;
-								return await this.db.createNotification(notification);
+			const { parse } = pkg;
+	
+			if (!req.file) {
+				return res.status(400).json({
+					errors: [
+						{ message: "No file uploaded" }
+					]
+				});
+			}
```

We should not return here, we should throw an error.  

We have error handling middleware that creates standard error responses and handles all errors.